src/os/bluestore/bluestore_tool.cc

```diff
@@ -572,6 +577,29 @@ int main(int argc, char **argv)
       exit(EXIT_FAILURE);
     }
   }
+  if (action == "trim") {
+    if (path.empty()) {
+      cerr << "must specify bluestore path" << std::endl;
+      exit(EXIT_FAILURE);
+    }
+    if (really.empty() || strcmp(really.c_str(), "--yes-i-really-really-mean-it") != 0) {
+      cerr << "Trimming is a dangerous operation which could cause data loss, "
+           << "please confirm with --yes-i-really-really-mean-it option"
+           << std::endl;
```

Comment:
I think we overdid with the warning message.
The risks are:
1) We implemented it wrong and we will now clear data. On that logic, everything should be considered dangerous.
In itself, trimming unused space poses little risk. That is, unless drive firmware will go crazy.
2) We will be trimming something that is actually in use. This is more likely. For it, we should have message like:
"Trimming a non healthy bluestore is a dangerous operation which could cause data loss,
please run fsck and confirm with -yes-i-really-really-mean-it option"

Reply:
got it

Reply:
3. The target drive does not [properly] support TRIM, esp. if the firmware is old.

Reply:
@anthonyeleven This is worrying. Do you have experience with drives that claimed to be able to do it, but have botched the implementation?
Do you think we should warn users to run fsck after trim?

Reply:
Yes, I have such experience.

I don't know that a fsck after trim will do much.  Some of the issues I've seen only happen with writes.

We can't predict what drives a user might have and how they behave, that's far beyond our scope.  But we should not have trimming happen by default, and we should caution users to approach it carefully for each of their drive SKUs and to update firmware to the latest available before trying.